pyqtgraph/widgets/MultiAxisPlotWidget.py

```diff
@@ -0,0 +1,427 @@
+        """
+        # add default axis to the list of axes if requested
+        if x_axis in ["bottom", "top"]:
+            x = self.addAxis(x_axis, axis=self.pi.axes[x_axis]["item"])
+        elif x_axis not in self.axes:
+            x = self.addAxis(x_axis)
+        else:
+            x = self.axes[x_axis]
+        if y_axis in ["left", "right"]:
+            y = self.addAxis(x_axis, axis=self.pi.axes[y_axis]["item"])
+        elif y_axis not in self.axes:
+            y = self.addAxis(y_axis)
+        else:
+            y = self.axes[y_axis]
+        if x_axis in ["bottom", "top"] and y_axis in ["left", "right"]:
+            # use default plotitem if none provided
+            plotitem = self.pi
+        else:
+            # VIEW
+            plotitem = PlotItem(parent=self.pi, name=name, *args,
+                                enableMenu=False, **kwargs)  # pass axisitems?
+            # hide all plotitem axis (they vould interfere with viewbox)
+            for a in ["left", "bottom", "right", "top"]:
+                plotitem.hideAxis(a)
+            # fix parent legend not showing child charts
+            plotitem.legend = self.pi.legend
+            for k, pos, axis in [["left", [2, 0], y], ["bottom", [3, 1], x], ["right", [2, 2], y], ["top", [1, 1], x]]:
+                # # DO NOT USE, WILL MAKE AXIS UNMATCHED TO DATA
+                # # you can't add the new ones after, it doesn't work for some reason
+                # # hide them instead
+                # plotitem.layout.removeItem(plotitem.layout.itemAt(*pos))
+                if axis.orientation == k:
+                    plotitem.axes[k] = {"item": axis, "pos": pos}
+                # hide plotitem axes
+                plotitem.hideAxis(a)
+        # CHART
+        if chart is None:
+            chart = PlotDataItem(name=name)
+        plotitem.addItem(chart)
+        # keep plotitem inside chart
+        chart.plotItem = plotitem
+        # keeptrack of connections
+        if chart.name not in self._signalConnectionsByChart:
+            self._signalConnectionsByChart[chart.name] = {}
+        # keep axis track
+        chart.axes = [x_axis, y_axis]
+        # keep chart
+        self.charts[name] = chart
+        # create a mapping for this chart and his axis
+        self.axes[x_axis].charts.append(name)
+        self.axes[y_axis].charts.append(name)
+        return chart, plotitem
+
+    def clearLayout(self):
+        """Clears the widget but keeps all created axis and charts."""
+        while self.layout.count() > 0:
+            item = self.layout.itemAt(0)
+            self.layout.removeAt(0)
+            self.scene().removeItem(item)
+            del item
+        for chart_name, chart in self.charts.items():
+            self.disconnect_all(chart)
+
+    def makeLayout(self, axes=None, charts=None):
+        """Adds all given axes and charts to the widget.
+
+        Parameters
+        ----------
+        axes : list of str or None:
```

Comment:
I messed you up here, instead of `or None` for these, it should be `, optional`  so in this case, it would be 

```
axes : list of str, optional
```

Reply:
done